### PR TITLE
Make backends of read_image and resize selectable

### PR DESCRIPTION
**Pull request by yuyu2172**

### CODE REVIEW

File: tests/transforms_tests/image_tests/test_resize.py [hunk 1]
```diff
@@ -3,25 +3,44 @@
...
+
+    if _cv2_available:
+        backend_params = {'backend': ['cv2', 'PIL']}
+    else:
+        backend_params = {'backend': ['PIL']}
+    params = testing.product_dict(params, testing.product(backend_params))
+    return params
+
+
+@testing.parameterize(*_create_paramters())
```
Comment (Hakuyume) [inline tests/transforms_tests/image_tests/test_resize.py]: Even if `_cv2_available = False`, `resize` should use `PIL` as fallback.
```
@testing.parameterize(*testing.product({
    'interpolation': [PIL.Image.NEAREST, PIL.Image.BILINEAR, PIL.Image.BICUBIC, PIL.Image.LANCZOS],
    'backend': ['cv2', PIL'],
}))
```

File: chainercv/utils/image/read_image.py [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
-    """Read an image from a file.
 
-    This function reads an image from given file. The image is CHW format and
-    the range of its value is :math:`[0, 255]`. If :obj:`color = True`, the
-    order of the channels is RGB.
+def _read_image_cv2(path, dtype, color):
+    if color:
+        color_option = cv2.IMREAD_COLOR
+    else:
+        color_option = cv2.IMREAD_GRAYSCALE
```
[the remark this answers is not in the document]
Comment (yuyu2172) [reply chainercv/utils/image/read_image.py]: I checked it to work at least for `CamVid`

File: tests/utils_tests/image_tests/test_read_image.py [hunk 3]
```diff
@@ -61,4 +65,39 @@ def test_read_image_mutable(self):
...
+    @unittest.skipUnless(_cv2_available, 'cv2 is not installed')
+    def test_read_image_different_backends_as_color(self):
+        chainer.config.cv_read_image_backend = 'cv2'
+        cv2_img = read_image(self.path, dtype=self.dtype, color=True)
+
+        chainer.config.cv_read_image_backend = 'PIL'
+        pil_img = read_image(self.path, dtype=self.dtype, color=True)
+
+        if self.suffix != 'jpg':
+            np.testing.assert_almost_equal(cv2_img, pil_img)
```
Comment (Hakuyume) [inline tests/utils_tests/image_tests/test_read_image.py]: Can we use `assert_equal` here? I hope they are same for bmp and png.

Comment (yuyu2172) [reply tests/utils_tests/image_tests/test_read_image.py]: thanks

**Pull request merged**